.github/workflows/test.yml

```diff
@@ -6,46 +6,89 @@ on:
 
 jobs:
   test:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            platform: linux-amd64
+            php-version: "8.3"
+            run-tests: true
+          - os: ubuntu-latest
+            platform: linux-arm64
+            php-version: "8.3"
+            run-tests: false
+          - os: macos-latest
+            platform: darwin-arm64
+            php-version: "8.3"
+            run-tests: true
+
+    name: ${{ matrix.run-tests && 'Test' || 'Build' }} on ${{ matrix.platform }} (PHP ${{ matrix.php-version }})
 
     steps:
       - name: Checkout code
         uses: actions/checkout@v5
 
+      - name: Install ARM64 cross-compilation tools
+        if: matrix.platform == 'linux-arm64'
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y gcc-aarch64-linux-gnu
+
       - name: Setup PHP
+        if: matrix.run-tests
         uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
 
       - name: Setup Go
         uses: actions/setup-go@v6
         with:
           go-version: "1.24"
 
-      - name: Build shared library
-        run: make build
+      - name: Build shared library for current platform
+        run: |
+          if [ "${{ matrix.platform }}" = "linux-arm64" ]; then
+            make build-linux-arm64
+          else
+            make build
+          fi
```

Comment:
Mmm, conceptually this shouldn't need two different branches, it builds the library for the current architecture, as the `name:` of the step correctly suggests.

We should `make build-${{ matrix.platform }}` here.